src/index.test.ts

```diff
@@ -0,0 +1,212 @@
+  const ts = outdent`
+    \`\`\`ts
+    import { ApolloServer } from 'apollo-server';
+    import typeDefs from './schema';
+    import preserved from 'preserved'; // preserve-line
+    import notPreserved from 'not-preserved';
+
+    const server = new ApolloServer({ typeDefs });
+    \`\`\`
+  `;
+
+  expect(
+    remark()
+      .use(remarkTypescript, {
+        prettierOptions: {
+          singleQuote: true
+        }
+      })
+      .processSync(ts)
+      .toString()
+  ).toEqual(
+    outdent`
+      \`\`\`ts
+      import { ApolloServer } from 'apollo-server';
+      import typeDefs from './schema';
+      import preserved from 'preserved';
+      import notPreserved from 'not-preserved';
+
+      const server = new ApolloServer({ typeDefs });
+      \`\`\`
+
+      \`\`\`js
+      import { ApolloServer } from 'apollo-server';
+      import typeDefs from './schema';
+      import preserved from 'preserved';
+
+      const server = new ApolloServer({ typeDefs });
+      \`\`\`
+
+    `
+  );
+});
+
+test('uses the skipping parameter', (): void => {
+  const ts = outdent`
+    \`\`\`ts
+    (): void => { }
+    \`\`\`
+
+    \`\`\`ts no-transpile
+    (): void => { }
+    \`\`\`
+
+    \`\`\`ts
+    (): void => { }
+    \`\`\`
+  `;
+  const result = remark()
+    .use(remarkTypescript, {
+      shouldSkip(_, parent, index, meta) {
+        if (meta?.includes('no-transpile')) {
+          return true;
+        }
+        const nextNode = parent.children[index + 1] as Code | undefined;
+        if (nextNode && nextNode.lang === 'ts') {
+          return true;
+        }
+        return false;
+      }
+    })
+    .processSync(ts)
+    .toString();
+
+  expect(result).toEqual(
+    outdent`
+      ${ts}
+
+      \`\`\`js
+      () => {};
+      \`\`\`
+
+    `
+  );
+});
+
+test('custom transfomration utilization', (): void => {
+  const ts = outdent`
+    \`\`\`ts {2}
+    type Result = void;
+    (): Result => { }
+    \`\`\`
+  `;
+  const result = remark()
+    .use(remarkTypescript, {
+      customTransformations: [
+        () => {
+          return {
+            afterTranspile(code, meta) {
+              const {highlight} = fenceparser(meta ?? '') ?? {};
+              const highlightedLines: number[] = [];
+              if (highlight) {
+                highlightedLines.push(
+                  ...rangeParser(Object.keys(highlight).toString())
+                );
+              }
+              return code
+                .split('\n')
+                .map((line, index) => {
+                  const isHighlighted = highlightedLines.includes(index + 1);
+                  if (!isHighlighted) {
+                    return line;
+                  }
+                  return line + ' // highlight-line';
+                })
+                .join('\n');
+            }
+          };
+        }
+      ]
+    })
+    .processSync(ts)
+    .toString();
+
+  expect(result).toEqual(
+    outdent`
+      ${ts}
```

Comment:
Should the TypeScript output after the transpile also have a `// highlight-line` comment?

````js
\`\`\`ts {2}
    type Result = void;
    (): Result => { } // highlight-line
\`\`\`

\`\`\`js {2}
      () => {}; // highlight-line
 \`\`\`

````

Reply:
Yeah, we would want the line highlighting comment to be preserved on both code blocks.

Reply:
With the way this is written, the highlight-line comment will be the only thing the `<MultiCodeBlock>` looks at for highlighting, ignoring the numbers entirely.

Reply:
I just noticed the TS code block still has the `{2}` in the metastring. If the JS codeblock had both a `{1}` in the metastring and a `// highlight-line` comment on the first line, then would the `{1}` be carried over to the TS code block, causing both the first and second lines to be highlighted?

Reply:
I noticed the `{2}` was preserved in the JS output in this example. Our custom transformer should probably also remove the highlight numbers from the metastring of the JS codeblock after it adds `highlight-line` comments.

If we're going to write and export our premade line highlighting transformer, it might be good to test that. Maybe in addition to this test with a simpler custom transformer.